From e161d3a1297408415842731be73af765fbdc5c3b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zbigniew=20J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Tue, 4 Jan 2022 09:31:25 +0100
Subject: [PATCH] fuzz-bcd: silence warning about always-true comparison

Occurs with gcc-11.2.1-7.fc35.x86_64.

(cherry picked from commit 5377ad4ea44e771b5eb436d381ea9f3506488295)

Related: #2087652
---
 src/boot/efi/fuzz-bcd.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/boot/efi/fuzz-bcd.c b/src/boot/efi/fuzz-bcd.c
index e5ed6638a4..3df55a5c36 100644
--- a/src/boot/efi/fuzz-bcd.c
+++ b/src/boot/efi/fuzz-bcd.c
@@ -21,6 +21,7 @@ int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
         assert_se(p);
 
         char16_t *title = get_bcd_title(p, size);
-        assert_se(!title || char16_strlen(title) >= 0);
+        if (title)
+                (void) char16_strlen(title);
         return 0;
 }
